File: mobile/src/main/java/com/gitlab/PCU/PCU/BackgroundService.java
package com.gitlab.PCU.PCU;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class BackgroundService extends Service {

    private LooperThread looperThread;

    public BackgroundService() {
        looperThread = new LooperThread();
    }

    public int onStartCommand(Intent intent, int flags, int startId) {
        looperThread.start();
        return START_REDELIVER_INTENT;
    }

    @Override
    public void onDestroy() {
        looperThread.interrupt();
        super.onDestroy();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: mobile/src/main/java/com/github/PCUnlocker/app/BackgroundService.kt
package com.github.PCUnlocker.app

import android.app.Service
import android.content.Intent
import android.os.IBinder

class BackgroundService : Service() {

    private var looperThread: LooperThread? = null

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        looperThread = LooperThread(applicationContext)
        looperThread!!.start()
        return Service.START_REDELIVER_INTENT
    }

    override fun onDestroy() {
        looperThread!!.interrupt()
        super.onDestroy()
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }
}
